File: python/counting_bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        result: List[int] = []

        for i in range(n + 1):
            count = 0
            num = i
            while num:
                count = count + num % 2
                num >>= 1
            result.append(count)

        return result

File: cpp/countingBits.cpp
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> countBits(int n) {
    vector<int> res;

    for (int i = 0; i < n + 1; i++) {
      int cur = 0;

      for (int j = 0; j < 32; j++) {
        if ((1 << j) & i) {
          cur++;
        }
      }
      res.push_back(cur);
    }
    return res;
  }
};
